Assets/Scripts/Player/PlayerInventory.cs

```diff
@@ -52,10 +57,12 @@ public override void Awake()
 	// Use this for initialization
 	void Start()
 	{
-		inventoryBar = GameObject.FindObjectOfType<InventoryScrollBar>();
-		DebugUtils.Assert( inventoryBar, "There must be an InventoryScrollBar object in the scene." );
+		_inventoryBar = GameObject.FindObjectOfType<InventoryScrollBar>();
+		DebugUtils.Assert( _inventoryBar, "There must be an InventoryScrollBar object in the scene." );
+
+		DayCycleManager.RegisterEndOfDayCallback( HideBackBuddy );
```

Comment:
Consider registering the reenable back buddy callback here as well. Might as well keep these two callbacks together